book-1-orientation/chapters/DATA_STRUCTURES_DICTIONARY.md

```diff
@@ -41,157 +41,230 @@ foreach(KeyValuePair<string, int> toy in toysSold)
+        part of speech: adjective
+        example sentence: I am excited to learn C#!
 */
-foreach (Dictionary<string, double> purchase in purchases)
+
+// Iterate the List of Dictionaries
+foreach ()
 {
-    foreach (KeyValuePair<string, double> stock in purchase)
+    // Iterate the KeyValuePairs of the Dictionary
+    foreach ()
     {
-        // Does the full company name key already exist in the `stockReport`?
-
-        // If it does, update the total valuation
-
-        /*
-            If not, add the new key and set its value.
-            You have the value of "GE", so how can you look
-            the value of "GE" in the `stocks` dictionary
-            to get the value of "General Electric"?
-        */
+        Console.WriteLine($"{wordData.Key}: {wordData.Value}");
     }
 }
 ```
 
-Now that the report dictionary is populated, display the final results.
+## Practice: English Idioms
+
+Create a new console application and paste the following code into your `Main()` method. Then write a `foreach` loop to produce the output in the image below.
 
 ```cs
-foreach(KeyValuePair<?, ?> item in stockReport)
-{
-    Console.WriteLine($"The position in {display the key} is worth {display the value}");
-}
+Dictionary<string, List<string>> idioms = new Dictionary<string, List<string>>();
+idioms.Add("Penny", new List<string> { "A", "penny", "for", "your", "thoughts" });
+idioms.Add("Injury", new List<string> { "Add", "insult", "to", "injury" });
+idioms.Add("Moon", new List<string> { "Once", "in", "a", "blue", "moon" });
+idioms.Add("Grape", new List<string> { "I", "heard", "it", "through", "the", "grapevine" });
+idioms.Add("Murder", new List<string> { "Kill", "two", "birds", "with", "one", "stone" });
+idioms.Add("Limbs", new List<string> { "It", "costs", "an", "arm", "and", "a", "leg" });
+idioms.Add("Grain", new List<string> { "Take","what","someone","says","with","a","grain","of","salt" });
+idioms.Add("Fences", new List<string> { "I'm", "on", "the", "fence", "about", "it" });
+idioms.Add("Sheep", new List<string> { "Pulled", "the", "wool", "over", "his", "eyes" });
+idioms.Add("Lucifer", new List<string> { "Speak", "of", "the", "devil" });
 ```
----
-# Practice: Iterating over planets
 
-Now we'll combine Dictionaries with the Lists we learned in the [previous chapter](./DATA_STRUCTURES_LIST.md).
+![idioms output](./images/idioms-output.png)
 
-## Instructions
+> Reference: [String.Join() method](https://docs.microsoft.com/en-us/dotnet/api/system.string.join?view=netframework-4.7.2)
+
+## Challenge: Randall's Car Lot in C#
+
+Take the following JavaScript data structure that represents car sales and convert it to C# Lists and Dictionaries.
+
+```js
+"vehicles": [
+    {
+        "vehicle": {
+            "year": 2008,
+            "model": "Damfresh",
+            "make": "Biotraxquote",
+            "color": "sky magenta"
+        },
+        "sales_id": "ecb1c841-1a43-4a7c-896e-712d2ec39c71",
+        "sales_agent": {
```

Comment:
I wouldn't use snake_case.

Reply:
That would be up to the students to make the variable names conform to C# conventions.  I'll add that to the instructions.